pinot-common/src/main/java/org/apache/pinot/common/response/broker/BrokerResponseNativeV2.java

```diff
@@ -77,20 +100,301 @@ public static BrokerResponseNativeV2 empty() {
+  }
+
+  @Override
+  public int getNumRowsResultSet() {
+    return 0;
+  }
+
+  @Override
+  public long getOfflineThreadCpuTimeNs() {
+    return _offlineThreadCpuTimeNs;
+  }
+
+  @Override
+  public long getRealtimeThreadCpuTimeNs() {
+    return _realtimeThreadCpuTimeNs;
+  }
+
+  @Override
+  public long getOfflineSystemActivitiesCpuTimeNs() {
+    return _offlineSystemActivitiesCpuTimeNs;
+  }
+
+  @Override
+  public long getRealtimeSystemActivitiesCpuTimeNs() {
+    return _realtimeSystemActivitiesCpuTimeNs;
+  }
+
+  @Override
+  public long getOfflineResponseSerializationCpuTimeNs() {
+    return _offlineResponseSerializationCpuTimeNs;
+  }
+
+  @Override
+  public long getRealtimeResponseSerializationCpuTimeNs() {
+    return _realtimeResponseSerializationCpuTimeNs;
+  }
+
+  @Override
+  public long getNumSegmentsPrunedByBroker() {
+    return _numSegmentsPrunedByBroker;
+  }
+
+  @Override
+  public void setNumSegmentsPrunedByBroker(long numSegmentsPrunedByBroker) {
+    _numSegmentsPrunedByBroker = numSegmentsPrunedByBroker;
+  }
+
+  @Override
+  public String getRequestId() {
+    return _requestId;
+  }
+
+  @Override
+  public void setRequestId(String requestId) {
+    _requestId = requestId;
+  }
+
+  @Override
+  public String getBrokerId() {
+    return _brokerId;
+  }
+
+  @Override
+  public void setBrokerId(String requestId) {
+    _brokerId = requestId;
+  }
+
+  @Override
+  public long getBrokerReduceTimeMs() {
+    return _brokerReduceTimeMs;
+  }
+
+  @Override
+  public void setBrokerReduceTimeMs(long brokerReduceTimeMs) {
+    _brokerReduceTimeMs = brokerReduceTimeMs;
+  }
+
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+  @Override
+  public boolean isPartialResult() {
+    return isNumGroupsLimitReached() || getExceptionsSize() > 0 || isMaxRowsInJoinReached();
+  }
+
+  public void addServerStats(StatMap<DataTable.MetadataKey> serverStats) {
+    // Set execution statistics.
+    _serverStats.merge(serverStats);
+
+    long threadCpuTimeNs = serverStats.getLong(DataTable.MetadataKey.THREAD_CPU_TIME_NS);
+    long systemActivitiesCpuTimeNs = serverStats.getLong(DataTable.MetadataKey.SYSTEM_ACTIVITIES_CPU_TIME_NS);
+    long responseSerializationCpuTimeNs = serverStats.getLong(DataTable.MetadataKey.RESPONSE_SER_CPU_TIME_NS);
+
+    String tableName = serverStats.getString(DataTable.MetadataKey.TABLE);
+    if (tableName != null) {
+      TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+
+      if (tableType == TableType.OFFLINE) {
+        _offlineThreadCpuTimeNs += threadCpuTimeNs;
+        _offlineSystemActivitiesCpuTimeNs += systemActivitiesCpuTimeNs;
+        _offlineResponseSerializationCpuTimeNs += responseSerializationCpuTimeNs;
+      } else {
+        _realtimeThreadCpuTimeNs += threadCpuTimeNs;
+        _realtimeSystemActivitiesCpuTimeNs += systemActivitiesCpuTimeNs;
+        _realtimeResponseSerializationCpuTimeNs += responseSerializationCpuTimeNs;
+      }
```

Comment:
This is not correct because tableName never contains the _OFFLINE or _REALTIME suffix, so these stats are always counted as realtime.

Right now as far as I can test, these stats are not being filled in V2. @Jackie-Jiang do you think we should just remove this or always count the times in either offline or realtime as we are doing right now?

Reply:
Good catch! I guess this is an example where we cannot directly expose V1 stats as V2 stats even for leaf stage. V1 server request only touch one table (either OFFLINE or REALTIME), but V2 leaf stage can touch 1 or 2 tables. To solve this, I think we should read the V1 stats in leaf stage, and convert them into leaf stage stats there.

Reply:
Oh, I think I misunderstood the issue, and we are already reposting the stats in leaf stage operator. And now I understand better why we need to make V1 stats also implementing `StatMap.Key`.
I guess we can just leave them as 0. People can lookup the V2 stats to get the details

Reply:
These are not V1 or V2 stats. They _broker_ stats. They are populated in the broker code.

V1 collects `threadCpuTimeNs`, `systemActivitiesCpuTimeNs` and `responseSerializationCpuTimeNs` but doesn't care whether they are realtime or offline. That information reaches the broker and depending on the table type, it stores the value in the realtime or offline version and that is what is returned to the customer.

In the code we have in master right now these stats are not being populated when the query is executed in V2. Specifically in `ExecutionStatsAggregator`, the code:

```java
    TableType tableType = null;
    String instanceName = null;
    if (routingInstance != null) {
      tableType = routingInstance.getTableType();
      instanceName = routingInstance.getShortName();;
    } else if (tableName != null) {
      tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
      instanceName = tableName;
    } else {
      tableType = null;
      instanceName = null;
    }
```

Ends up setting `tableType` to null, these stats are never calculated.

I guess we can remove these lines and keep the current behavior. We can try to add support for this in the future.